node/pkg/solana/client.go

```diff
@@ -155,6 +158,17 @@ func (s *SolanaWatcher) Run(ctx context.Context) error {
 				for _, acc := range accs {
 					s.fetchMessageAccount(rCtx, logger, solana.MustPublicKeyFromBase58(acc), 0)
 				}
+			case m := <-s.obsvReqC:
+				if m.ChainId != uint32(vaa.ChainIDSolana) {
+					panic("unexpected chain id")
```

Comment:
It panics because we have an earlier check here:

https://github.com/certusone/wormhole/blob/fae78bf8281b292556167f84e51833c8faeaa8ca/node/pkg/p2p/p2p.go#L496-L498